src/libraries/System.Text.Json/tests/Serialization/CyclicTests.cs

```diff
@@ -21,35 +21,43 @@ public static void WriteCyclicFailDefault()
         }
 
         [Theory]
-        [InlineData(1)]
-        [InlineData(2)]
-        [InlineData(10)]
-        [InlineData(70)]
-        public static void WriteCyclicFail(int depth)
+        [InlineData(1, 2)]
+        [InlineData(2, 3)]
+        [InlineData(10, 11)]
+        [InlineData(70, 71)]
+        [InlineData(10, 0)]
+        public static void WriteCyclic(int objectHierarchyDepth, int maxDepth)
         {
             var rootObj = new TestClassWithCycle("root");
-            CreateObjectHierarchy(1, depth, rootObj);
+            CreateObjectHierarchy(1, objectHierarchyDepth, rootObj);
 
-            {
-                var options = new JsonSerializerOptions();
-                options.MaxDepth = depth + 1;
+            var options = new JsonSerializerOptions();
+            options.MaxDepth = maxDepth;
 
-                // No exception since depth was not greater than MaxDepth.
-                string json = JsonSerializer.Serialize(rootObj, options);
-                Assert.False(string.IsNullOrEmpty(json));
-            }
+            string json = JsonSerializer.Serialize(rootObj, options);
+            Assert.False(string.IsNullOrEmpty(json));
+        }
 
-            {
-                var options = new JsonSerializerOptions();
-                options.MaxDepth = depth;
-                JsonException ex = Assert.Throws<JsonException>(() => JsonSerializer.Serialize(rootObj, options));
-
-                // Exception should contain the path and MaxDepth.
-                // Since the last Parent property is null, the serializer moves onto the Children property.
-                string expectedPath = "$" + string.Concat(Enumerable.Repeat(".Parent", depth - 1));
-                Assert.Contains(expectedPath, ex.Path);
-                Assert.Contains(depth.ToString(), ex.ToString());
-            }
+        [Theory]
+        [InlineData(1, 1, 0)]
+        [InlineData(2, 2, 1)]
+        [InlineData(10, 10, 9)]
+        [InlineData(70, 70, 69)]
+        [InlineData(70, 0, 63)]
```

Comment:
Same nit: comment that 0 is special and means 64.

Reply:
fixed in 70835c4